blockchain/timestamping.py

```diff
@@ -0,0 +1,211 @@
+compliance with the Apache License and the following modification to it:
+Section 6. Trademarks. is deleted and replaced with:
+
+     6. Trademarks. This License does not grant permission to use the trade
+        names, trademarks, service marks, or product names of the Licensor
+        and its affiliates, except as required to comply with Section 4(c) of
+        the License and to reproduce the content of the NOTICE file.
+
+You may obtain a copy of the Apache License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the Apache License with the above modification is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied. See the Apache License for the specific
+language governing permissions and limitations under the Apache License.
+"""
+
+__author__ = "Joe Roets, Brandon Kite, Dylan Yelton, Michael Bachtel"
+__copyright__ = "Copyright 2016, Disney Connected and Advanced Technologies"
+__license__ = "Apache"
+__version__ = "2.0"
+__maintainer__ = "Joe Roets"
+__email__ = "joe@dragonchain.org"
+
+
+"""
+Module for storing data into the bitcoin blockchain transaction ontput scripts
+"""
+
+import bitcoin.rpc
+
+from bitcoin import params
+from bitcoin.core import *
+from bitcoin.core.script import *
+
+import requests
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class IPoEStore(object):
+    """
+    Abstracts a proof-of-existence store.
+    """
+
+    def ispersisted(self, id, data):
+        """
+        Verifies ``data`` item is persisted on the datastore with identifier ``id``
+        Args:
+            id: the data item identifier within the datastore
+            data: the item data we want to verify
+
+        Returns:
+            True if the ``data`` item is in the datastore with identifier ``id``; otherwise False
+        """
+        pass
+
+    def persist(self, data):
+        """
+        Persists data in a datastore and returns and identifier for future reference
+        Args:
+            data:   the item data to be persisted
+
+        Returns:
+            An identifier for referencing the saved data item:
+        """
+        pass
+
+
+class BitcoinTimestamper(): # IPoEStore
+    """
+    Stamps a data item in the bitcoin blockchain and also allows us to verifies the
+    existence of a data item in a given bitcoin transaction output. For doing this
+    a bitcoin node is required and available through RPC. The access data such as
+    user name and password are read from ~/.bitcoin/bitcoin.conf configuration file
+
+    usage example:
+        fee_provider = BitcoinFeeProvider()
+        stamper = BitcoinTimestamper("regtest", fee_provider)
+        txid = stamper.persist("hello world!")
+        assert stamper.ispersisted(txid, "hello world!"), "data item not present in the given txid"
+    """
+    MIN_FEE_BYTE = 60
+    MAX_FEE_BYTE = 250
```

Comment:
Yes - let's @wolfpack94 make a ticket to make these (and maybe other items in the Bitcoin bridge) configuration items.

For example, the fee calculation provider class (I'd like to rename the implemented provider from `BitcoinFeeProvider` to `21coBitcoinFeeProvider).